quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -0,0 +1,207 @@
+        case QK_LAYER_TAP_TOGGLE ... QK_LAYER_TAP_TOGGLE_MAX:
+            if ((keycode & 0xff) == auto_mouse.config.layer) {
+                if (record->event.pressed) {
+                    auto_mouse.status.mouse_key_tracker++;
+                } else {
+#if TAPPING_TOGGLE != 0 && !defined(NO_ACTION_TAPPING)
+                    if (record->tap.count == TAPPING_TOGGLE) {
+                        toggle_mouse_layer();
+#    if TAPPING_TOGGLE == 1
+                        if (!auto_mouse.status.layer_toggled) auto_mouse.status.mouse_key_tracker -= record->tap.count + 1;
+#    else
+                        if (!auto_mouse.status.layer_toggled) auto_mouse.status.mouse_key_tracker -= record->tap.count;
+#    endif
+                    } else {
+                        auto_mouse.status.mouse_key_tracker--;
+                    }
+#else
+                    auto_mouse.status.mouse_key_tracker--;
+#endif
+                }
+                auto_mouse.timer.active = timer_read();
+            }
+            break;
+        // LM(auto_mouse.config.layer, mod)------------------------------------------------------------
+        case QK_LAYER_MOD ... QK_LAYER_MOD_MAX:
+            if (((keycode >> 4) & 0xF) == auto_mouse.config.layer) {
+                record->event.pressed ? auto_mouse.status.mouse_key_tracker++ : auto_mouse.status.mouse_key_tracker--;
+                auto_mouse.timer.active = timer_read();
+            }
+            break;
+#ifndef NO_ACTION_TAPPING
+        case QK_MOD_TAP ... QK_MOD_TAP_MAX:
+            if (record->event.pressed || !record->tap.count) {
+                break;
+            }
+#endif
+        default:
+            if (IS_NOEVENT(record->event)) break; // skip if no key has been pressed
+            if (is_mouse_record(keycode, record)) {
+                record->event.pressed ? auto_mouse.status.mouse_key_tracker++ : auto_mouse.status.mouse_key_tracker--;
+                auto_mouse.timer.active = timer_read();
+            } else {
+                // turn off mouse layer if non mouse key is pressed and start/restart debounce timer
+                if (layer_state_is(auto_mouse.config.layer) && !auto_mouse.status.mouse_key_tracker && !auto_mouse.status.layer_toggled) {
+                    layer_off(auto_mouse.config.layer);
+                }
+                auto_mouse.timer.active             = 0;
+                auto_mouse.status.mouse_key_tracker = 0;
+                auto_mouse.timer.delay              = timer_read();
+            }
+    }
+}
+
+/* Callbacks for adding keyrecords as mouse keys */
+__attribute__((weak)) bool is_mouse_record_kb(uint16_t keycode, keyrecord_t* record) {
+    return is_mouse_record_user(keycode, record);
+}
+__attribute__((weak)) bool is_mouse_record_user(uint16_t keycode, keyrecord_t* record) {
+    return is_mouse_record_keymap(keycode, record);
+}
+__attribute__((weak)) bool is_mouse_record_keymap(uint16_t keycode, keyrecord_t* record) {
+    return false;
+}
```

Comment:
```suggestion
}
#endif // POINTING_DEVICE_AUTO_MOUSE_ENABLE
```

Reply:
Huh I missed this but now this should be good